security/providers/oidc/src/main/java/io/helidon/security/providers/oidc/TenantAuthenticationHandler.java

```diff
@@ -394,10 +396,19 @@ private AuthenticationResponse errorResponse(ProviderRequest providerRequest,
                     + "nonce=" + nonce + "&"
                     + "state=" + state;
 
+            if (oidcConfig.pkceEnabled()) {
+                PkceChallengeMethod pkceChallengeMethod = oidcConfig.pkceChallengeMethod();
+                String codeChallenge = pkceChallengeMethod.transform(pkceVerifier);
+                queryString += "&"
+                        + "code_challenge=" + codeChallenge + "&"
+                        + "code_challenge_method=" + pkceChallengeMethod.method();
+            }
+
             JsonObject stateJson = JSON_BUILDER_FACTORY.createObjectBuilder()
                     .add("originalUri", origUri)
                     .add("state", state)
                     .add("nonce", nonce)
+                    .add("pkceVerifier", pkceVerifier)
```

Comment:
Is it really fine to add it even when not configured? Is there a chance this may break existing usages?

Reply:
This is just our state cookie. It does not break any existing usages. It will be used only if it was enabled and if it was enabled, it would have been configured.